src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2149,24 +2187,26 @@ private bool HasImplicitConversionToInterface(TypeSymbol source, TypeSymbol dest
 
             // * From any class type S to any interface type T provided S implements an interface
             //   convertible to T.
+            if (source.IsClassType())
+            {
+                return HasAnyBaseInterfaceConversion(source, destination, ref useSiteDiagnostics);
+            }
+
             // * From any interface type S to any interface type T provided S implements an interface
             //   convertible to T.
             // * From any interface type S to any interface type T provided S is not T and S is 
             //   an interface convertible to T.
-
-            if (source.IsClassType() && HasAnyBaseInterfaceConversion(source, destination, ref useSiteDiagnostics))
+            if (source.IsInterfaceType())
             {
-                return true;
-            }
-
-            if (source.IsInterfaceType() && HasAnyBaseInterfaceConversion(source, destination, ref useSiteDiagnostics))
-            {
-                return true;
-            }
+                if (HasAnyBaseInterfaceConversion(source, destination, ref useSiteDiagnostics))
+                {
+                    return true;
+                }
 
-            if (source.IsInterfaceType() && source != destination && HasInterfaceVarianceConversion(source, destination, ref useSiteDiagnostics))
-            {
-                return true;
+                if (!HasIdentityConversionInternal(source, destination, IncludeNullability) && HasInterfaceVarianceConversion(source, destination, ref useSiteDiagnostics))
```

Comment:
>IncludeNullability [](start = 72, length = 18)

Can we drop this argument?

Reply:
No. `HasImplicitConversionToInterface` should return `true` when `source` is `IEnumerable<object>` and `destination` is `IEnumerable<object?>` when considering nullability and should return `false when ignoring nullability.

---
In reply to: [168041099](https://github.com/dotnet/roslyn/pull/23981#discussion_r168041099) [](ancestors = 168041099)